core/src/main/java/com/radixdlt/messaging/MessagingModule.java

```diff
@@ -126,102 +129,124 @@ protected void configure() {
-        StatusRequest.class, messageCentralLedgerSync.statusRequestDispatcher());
+        NodeId.class, StatusRequest.class, messageCentralLedgerSync.statusRequestDispatcher());
   }
 
   @ProvidesIntoSet
-  private RxRemoteDispatcher<?> statusResponseDispatcher(
+  private RxRemoteDispatcher<?, ?> statusResponseDispatcher(
       MessageCentralLedgerSync messageCentralLedgerSync) {
     return RxRemoteDispatcher.create(
-        StatusResponse.class, messageCentralLedgerSync.statusResponseDispatcher());
+        NodeId.class, StatusResponse.class, messageCentralLedgerSync.statusResponseDispatcher());
   }
 
   @ProvidesIntoSet
-  private RxRemoteDispatcher<?> pingDispatcher(
+  private RxRemoteDispatcher<?, ?> pingDispatcher(
       MessageCentralPeerLiveness messageCentralPeerLiveness) {
-    return RxRemoteDispatcher.create(Ping.class, messageCentralPeerLiveness.pingDispatcher());
+    return RxRemoteDispatcher.create(
+        NodeId.class, Ping.class, messageCentralPeerLiveness.pingDispatcher());
   }
 
   @ProvidesIntoSet
-  private RxRemoteDispatcher<?> pongDispatcher(
+  private RxRemoteDispatcher<?, ?> pongDispatcher(
       MessageCentralPeerLiveness messageCentralPeerLiveness) {
-    return RxRemoteDispatcher.create(Pong.class, messageCentralPeerLiveness.pongDispatcher());
+    return RxRemoteDispatcher.create(
+        NodeId.class, Pong.class, messageCentralPeerLiveness.pongDispatcher());
   }
 
   @ProvidesIntoSet
-  private RxRemoteDispatcher<?> getPeersDispatcher(
+  private RxRemoteDispatcher<?, ?> getPeersDispatcher(
       MessageCentralPeerDiscovery messageCentralPeerDiscovery) {
     return RxRemoteDispatcher.create(
-        GetPeers.class, messageCentralPeerDiscovery.getPeersDispatcher());
+        NodeId.class, GetPeers.class, messageCentralPeerDiscovery.getPeersDispatcher());
   }
 
   @ProvidesIntoSet
-  private RxRemoteDispatcher<?> peersResponseDispatcher(
+  private RxRemoteDispatcher<?, ?> peersResponseDispatcher(
       MessageCentralPeerDiscovery messageCentralPeerDiscovery) {
     return RxRemoteDispatcher.create(
-        PeersResponse.class, messageCentralPeerDiscovery.peersResponseDispatcher());
+        NodeId.class, PeersResponse.class, messageCentralPeerDiscovery.peersResponseDispatcher());
+  }
+
+  @ProvidesIntoSet
+  private RxRemoteDispatcher<?, ?> ledgerStatusUpdateDispatcher(
+      MessageCentralLedgerSync messageCentralLedgerSync) {
+    return RxRemoteDispatcher.create(
+        NodeId.class,
+        LedgerStatusUpdate.class,
+        messageCentralLedgerSync.ledgerStatusUpdateDispatcher());
   }
 
   @ProvidesIntoSet
-  private RxRemoteDispatcher<?> ledgerStatusUpdateDispatcher(
+  private RxRemoteDispatcher<?, ?> bftLedgerStatusUpdateDispatcher(
       MessageCentralLedgerSync messageCentralLedgerSync) {
     return RxRemoteDispatcher.create(
-        LedgerStatusUpdate.class, messageCentralLedgerSync.ledgerStatusUpdateDispatcher());
+        BFTValidatorId.class,
```

Comment:
This appears to be the only place we register with `N = BFTValidatorId`, but it appears to duplicate the above which has `N = NodeId`. So it looks like each remote message can be registered with both?

I'm really not sure I understand the model here, maybe worth a discussion on slack?